kgtk/cli/graph_embeddings.py

```diff
@@ -0,0 +1,452 @@
+
+        input_kgtk_file: Path = kwargs['input_file_path']
+        tmp_folder = kwargs['temporary_directory']
+        tmp_tsv_path:Path = tmp_folder / f'tmp_{input_kgtk_file.name}'
+        # tmp_tsv_path:Path = input_kgtk_file.parent/f'tmp_{input_kgtk_file.name}'
+
+        #  make sure the tmp folder exists, otherwise it will raise an exception
+        if not os.path.exists(tmp_folder):
+            os.makedirs(tmp_folder)
+
+        output_kgtk_file = kwargs['output_file_path']
+        try:   #if output_kgtk_file is not empty, delete it
+            output_kgtk_file.unlink() 
+        except: pass # didn't find, then let it go
+
+        # *********************************************
+        # 0. PREPARE PBG TSV FILE
+        # *********************************************
+        reader_options: KgtkReaderOptions = KgtkReaderOptions.from_dict(kwargs)
+        value_options: KgtkValueOptions = KgtkValueOptions.from_dict(kwargs)
+        error_file: typing.TextIO = sys.stdout if kwargs.get("errors_to_stdout") else sys.stderr
+        kct: KgtkCopyTemplate = KgtkCreateTmpTsv(
+            input_file_path=input_kgtk_file,
+            output_file_path=tmp_tsv_path,
+            reader_options=reader_options,
+            value_options=value_options,
+            error_file=error_file,
+            verbose=verbose,
+            very_verbose=very_verbose,
+        )
+        # prepare the graph file
+        # create a tmp tsv file for PBG embedding
+        
+        logging.info('Generate the valid tsv format for embedding ...')
+        kct.process()
+        logging.info('Embedding file is ready...')
+        
+        # *********************************************
+        # 1. DEFINE CONFIG  
+        # *********************************************
+        raw_config = get_config(**kwargs)
+
+        ## setting corresponding learning rate and loss function for different algorthim
+        processed_config = config_preprocess(raw_config)
+        
+        # temporry output folder 
+        tmp_output_folder = Path(processed_config['entity_path'])
+
+        # before moving, need to check whether the tmp folder is not empty in case of bug
+        try: #if temporry output folder is alrady existing then delete it
+            shutil.rmtree(tmp_output_folder)
+        except: pass # didn't find, then let it go
+
+        # **************************************************
+        # 2. TRANSFORM GRAPH TO A BIGGRAPH-FRIENDLY FORMAT
+        # **************************************************
+        setup_logging()
+        config = parse_config(processed_config)
+        subprocess_init = SubprocessInitializer()
+        input_edge_paths = [tmp_tsv_path] 
+
+        convert_input_data(
+            config.entities,
+            config.relations,
+            config.entity_path,
+            config.edge_paths,
+            input_edge_paths,
+            TSVEdgelistReader(lhs_col=0, rel_col=1, rhs_col=2),
```

Comment:
this line assumes that the columns are in position 0, 1, 2 but this might not be the case

Reply:
Yeah, that's fine, cause we use the KGTKReader to read the nodes and relation, and we do an intermediate process to make the node1 at position 0, relation at position 1, and node2 at position 2